app/models/project.rb

```diff
@@ -80,6 +80,19 @@ class Project < ApplicationRecord
     joins(:pull_requests).where('pull_requests.merged_at > ?', date)
   }
 
+  scope :by_department, lambda { |department|
+    joins(language: :department).where(departments: { id: department.id })
+  }
+
+  scope :by_metrics_time, lambda { |from|
+    joins(:code_climate_project_metric)
```

Comment:
if you want projects without cc, then does it make sense to join code_climate_project_metric table?